lib/rules/no-warning-comments.js

```diff
@@ -79,12 +81,12 @@ module.exports = {
                  */
                 prefix = "^\\s*";
             } else if (/^\w/.test(term)) {
-                prefix = "\\b";
+                prefix = wordBoundary;
             } else {
                 prefix = "";
             }
 
-            return new RegExp(prefix + escaped + suffix, "i");
+            return new RegExp(prefix + escaped + suffix + eitherOrWordBoundary + term + wordBoundary, "i");
```

Comment:
I think it would be really helpful to have a comment here showing what the regex should look like for an example term.

Also: Based on the issues I've identified in the tests, we want to make sure that this regex is only augmented when `{ location: "anywhere" }` option is specified.